symfony/plugins/orangehrmAdminPlugin/Dao/WorkShiftDao.php

```
@@ -19,29 +19,73 @@
+            if ($workShift instanceof WorkShift) {
+                return $workShift;
+            }
+            return null;
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
+        }
     }
 
     /**
-     * @param int $workShiftId
-     * @return WorkShift|null
+     * @param WorkShiftSearchFilterParams $workShiftSearchFilterParams
+     * @return array
+     * @throws DaoException
      */
-    public function getWorkShiftById(int $workShiftId): ?WorkShift
+    public function getWorkShiftList(WorkShiftSearchFilterParams $workShiftSearchFilterParams): array
+    {
+        try {
+            $paginator = $this->getWorkShiftListPaginator($workShiftSearchFilterParams);
+            return $paginator->getQuery()->execute();
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
+        }
+    }
+
+    /**
+     * @param WorkShiftSearchFilterParams $workShiftSearchFilterParams
+     * @return Paginator
+     */
+    public function getWorkShiftListPaginator(WorkShiftSearchFilterParams $workShiftSearchFilterParams
+    ): Paginator {
+        $q = $this->createQueryBuilder(WorkShift::class, 'ws');
+        $this->setSortingAndPaginationParams($q, $workShiftSearchFilterParams);
+        return new Paginator($q);
+    }
+
+
+    /**
+     * @param WorkShiftSearchFilterParams $workShiftSearchFilterParams
+     * @return int
+     * @throws DaoException
+     */
+    public function getWorkShiftCount(WorkShiftSearchFilterParams $workShiftSearchFilterParams): int
     {
-        return $this->getRepository(WorkShift::class)->find($workShiftId);
+        try {
+            $paginator = $this->getWorkShiftListPaginator($workShiftSearchFilterParams);
+            return $paginator->count();
+        } catch (Exception $e) {
+            throw new DaoException($e->getMessage(), $e->getCode(), $e);
```

There is no use of catch-all exception and wrap as Dao exception  .. we will change this pattern globally